src/appliedsciencestudio/covid19_clj_viz/italia.clj

```diff
@@ -51,51 +53,70 @@
     "Emilia Romagna" "Emilia-Romagna"
     "Valle d'Aosta" "Valle d'Aosta/Vallée d'Aoste"} region))
 
+;; TODO: This is a nasty, unexpected edge case that would require a refactor to handle correctly
 (defn normalize-trentino
-  "The COVID numbers from Italy split one region into two. 'P.A. Bolzano' and 'P.A. Trento' should be combined into 'Trentino-Alto Adige/Südtirol.'"
+  "REGION: The COVID numbers from Italy split one region into two. 'P.A. Bolzano' and 'P.A. Trento' should be combined into 'Trentino-Alto Adige/Südtirol.'"
   [region-covid-data]
   (let [keys-to-sum [:hospitalized :icu :tot-hospitalized :quarantined :tot-positives :new-positives :recovered :dead :cases :tests]
         regions-to-combine (filter #(or (= "P.A. Bolzano" (:region-name %)) (= "P.A. Trento" (:region-name %))) region-covid-data)
         name-combined-region (fn [combined-regions] (assoc combined-regions :region-name "Trentino-Alto Adige/Südtirol"))
         remove-inexact-data (fn [combined-regions] (dissoc combined-regions :lat :lon))]
-    (->> (map #(select-keys % keys-to-sum) regions-to-combine)
-         (reduce #(merge-with + %1 %2))
-         (conj (first regions-to-combine))
-         (name-combined-region)
-         (remove-inexact-data)
-         (conj region-covid-data))))
-
-(def region-covid-data
-  (->> (mcsv/read-csv "resources/Italia-COVID-19/dati-regioni/dpc-covid19-ita-regioni-latest.csv"
-                      {:field-names-fn fields-it->en})
-       (map #(update % :region-name (fn [region-name]
-                                      (if-let [update-region-name (normalize-region-names region-name)]
-                                        update-region-name
-                                        region-name))))
-       (normalize-trentino)))
+    (->> (map #(select-keys % keys-to-sum) regions-to-combine) ; grab the important keys from the regions we want to combine
+         (reduce #(merge-with + %1 %2))                        ; add the info from each key together
+         (conj (first regions-to-combine))                     ; add in the rest of the information (name, region number, etc...)
+         (name-combined-region)                                ; name the combined region
+         (remove-inexact-data)                                 ; cleanup: remove longitude and latitude because it no longer applies
```

Comment:
These `let`-bound functions would probably read better if inlined. You can keep the names. For example, just for `remove-inexact-data`:

```clojure
(->> (map #(select-keys % keys-to-sum) regions-to-combine)
         (reduce #(merge-with + %1 %2))
         (conj (first regions-to-combine))
         (name-combined-region)
         (fn remove-inexact-data [combined-regions] (dissoc combined-regions :lat :lon))
         (conj region-covid-data))
```

Reply:
Great idea. This whole function is a bit embarrassing. I painted myself in a corner after not really thinking enough about how I'm going to conform the three data sources (covid, geography, and population).